### PR TITLE
Auto-style swift sources

### PR DESCRIPTION
**Pull request by wilhuff**
Format swift sources using [SwiftFormat](https://github.com/nicklockwood/SwiftFormat) when using `scripts/style.sh`. This fixes https://github.com/firebase/firebase-ios-sdk/issues/839.

Note SwiftFormat also fixes a few common coding problems. Most notably, it removes `weak` qualifiers from IBOutlets -- since iOS 6 this is no longer recommended.

The tool is imperfect, unfortunately:
  * It doesn't implement line wrapping at a fixed length
  * It fails when sorting imports (I've disabled this)

However, something is better than nothing and the result is largely in-line with Google's swift style guide.

This change adds a requirement that you have `swiftformat` installed if you're building on macOS.

### CODE REVIEW

File: scripts/style.sh [hunk 1]
```diff
@@ -21,19 +21,61 @@
...
   echo "If it's installed via homebrew you can run: brew upgrade clang-format"
   exit 1
 fi
 
-if [[ $# -gt 0 && "$1" = "test-only" ]]; then
+if [[ "$system" == "Darwin" ]]; then
+  version=$(swiftformat --version)
+  version="${version/*version /}"
+  # Allow an older swiftformat because travis isn't running high sierra yet
+  # and the formula hasn't been updated in a while on sierra :-/.
```
Comment (paulb777) [inline scripts/style.sh]: FYI, we can force High Sierra by changing to osx_image: xcode9.3beta. See https://docs.travis-ci.com/user/reference/osx/. Probably better not to run beta's on travis though.

File: Example/tvOSSample/tvOSSample/AuthLoginViewController.swift [hunk 2]
```diff
@@ -16,25 +16,22 @@ import UIKit
...
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destinationViewController.
-        // Pass the selected object to the new view controller.
-    }
-    */
+  // // In a storyboard-based application, you will often want to do a little preparation before navigation
+  // override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+  //   // Get the new view controller using segue.destinationViewController.
+  //   // Pass the selected object to the new view controller.
+  // }
```
Comment (ryanwilson) [inline Example/tvOSSample/tvOSSample/AuthLoginViewController.swift]: This seems to have formatted poorly, but let's just remove it (I can do so in a follow up PR).

File: scripts/style.sh [hunk 3]
```diff
@@ -21,19 +21,61 @@
...
   echo "Please upgrade to clang-format version 6."
   echo "If it's installed via homebrew you can run: brew upgrade clang-format"
   exit 1
 fi
 
-if [[ $# -gt 0 && "$1" = "test-only" ]]; then
+if [[ "$system" == "Darwin" ]]; then
+  version=$(swiftformat --version)
+  version="${version/*version /}"
+  # Allow an older swiftformat because travis isn't running high sierra yet
```
Comment (ryanwilson) [inline scripts/style.sh]: Nit: please capitalize High Sierra and Sierra

File: scripts/style.sh [hunk 4]
```diff
@@ -21,19 +21,61 @@
...
+  # and the formula hasn't been updated in a while on sierra :-/.
+  if [[ "$version" != "0.32.0" && "$version" != "0.33"* ]]; then
+    echo "Please upgrade to swiftformat 0.33.3"
+    echo "If it's installed via homebrew you can run: brew upgrade swiftformat"
+    exit 1
+  fi
+fi
+
+# Joins the given arguments with the separator given as the first argument
```
Comment (ryanwilson) [inline scripts/style.sh]: Nit: Period at the end.

File: scripts/style.sh [hunk 5]
```diff
@@ -21,19 +21,61 @@
...
+function join() {
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+clang_options=(-style=file)
+
+# Rules to disable in swiftformat
```
Comment (ryanwilson) [inline scripts/style.sh]: Nit: punctuation (colon or period?)

File: scripts/style.sh [hunk 6]
```diff
@@ -21,19 +21,61 @@
...
+swift_disable=(
+  # sortedImports is broken, sorting into the middle of the copyright notice
```
Comment (ryanwilson) [inline scripts/style.sh]: Nit: Period at the end.

File: scripts/style.sh [hunk 7]
```diff
@@ -21,19 +21,61 @@
...
+  sortedImports
+
+  # Too many of our swift files have simplistic examples. While technically
+  # it's correct to remove the unused argument labels, it makes our examples
+  # look wrong.
+  unusedArguments
+)
+
+swift_options=(
+  # Mimic Objective-C style
```
Comment (ryanwilson) [inline scripts/style.sh]: Nit: Period.

File: Example/tvOSSample/tvOSSample/AuthLoginViewController.swift (hunk 2 above)
Comment (wilhuff) [reply Example/tvOSSample/tvOSSample/AuthLoginViewController.swift]: This was previously commented out with a single `/*` and a matching `*/`. This ended formatting even worse. I've removed this for now.

File: scripts/style.sh (hunk 3 above)
Comment (wilhuff) [reply scripts/style.sh]: Done.

File: scripts/style.sh (hunk 1 above)
Comment (wilhuff) [reply scripts/style.sh]: Agreed. Let's leave it for now.

File: scripts/style.sh (hunk 4 above)
Comment (wilhuff) [reply scripts/style.sh]: Done.

File: scripts/style.sh (hunk 5 above)
Comment (wilhuff) [reply scripts/style.sh]: Done.

File: scripts/style.sh (hunk 6 above)
Comment (wilhuff) [reply scripts/style.sh]: Done.

File: scripts/style.sh (hunk 7 above)
Comment (wilhuff) [reply scripts/style.sh]: Done.

**Pull request merged**